### PR TITLE
cargo clippy integration: major part of auto fixes

### PR DESCRIPTION
**Pull request by s373r**
## Description

Related issue: https://github.com/kamu-data/kamu-cli/issues/332

## Checklist before requesting a review
- [x] [CHANGELOG.md](./CHANGELOG.md) updated
- [x] API changes are backwards-compatible
- [x] Workspace layout changes include a migration
- [x] Documentation update PR: <link or N/A>
- [x] Dataset pipelines update scheduled if needed

### CODE REVIEW

File: src/adapter/graphql/src/scalars/odf_generated.rs [hunk 1]
```diff
@@ -85,8 +85,8 @@ pub struct AttachmentEmbedded {
 impl From<odf::AttachmentEmbedded> for AttachmentEmbedded {
     fn from(v: odf::AttachmentEmbedded) -> Self {
         Self {
-            path: v.path.into(),
-            content: v.content.into(),
+            path: v.path,
```
Comment (sergiimk) [inline src/adapter/graphql/src/scalars/odf_generated.rs:88]: Note that you are changing a code-generated file

**Review by sergiimk — COMMENTED** [review]
Changes look pretty straightforward, just need to figure out what to do for codegen files. :+1: 

Few questions:
1) Is this all fixes or more is coming?
2) Should we add a clippy run to `make lint` target?
3) Should we run it in CI
4) Is there a way to enable clippy warnings in VSCode and if yes should we mention it in `DEVELOPER.md`

Note: I'm a little worried about diff explosion with my `breaking-changes` branch, so if there will be more radical changes coming - let's postpone them after the merge.

**Comment by s373r** [comment]
@sergiimk ,

> Changes look pretty straightforward, just need to figure out what to do for codegen files. :+1: 

I have already some ideas for that (uncommited), will be implemented in the follow-up PRs (just to reduce diffs and cognitive attack to reviewers)

> Few questions:
> 1) Is this all fixes or more is coming?
yes

> 2) Should we add a clippy run to `make lint` target?
for sure, will be added

> 3) Should we run it in CI
in the separate PR

> 4) Is there a way to enable clippy warnings in VSCode and if yes should we mention it in `DEVELOPER.md`
will investigate that

> Note: I'm a little worried about diff explosion with my `breaking-changes` branch, so if there will be more radical changes coming - let's postpone them after the merge.
I'm preparing a base for automatic changes, so, we can choose different strategies if there are a lot of merge conflicts -- for example, I can extract and provide changes (patches, diffs) to run linting and fixes that you apply to your branch (for example)

File: src/adapter/graphql/src/scalars/odf_generated.rs (hunk 1 above)
Comment (s373r) [reply src/adapter/graphql/src/scalars/odf_generated.rs:88]: I'll exclude this file for now -- a good catch! :medal_sports:

**Review by sergiimk — APPROVED** [review]

**Pull request merged**